ESP hosted: esp_hosted_status function signature invalid.

### Describe the bug

At some point in the past a bunch of esp_hosted APIs changed and were updated except for esp_hosted_status, which has its function signature missmatching.

```
zephyr/drivers/wifi/esp_hosted/esp_hosted_wifi.c:651:25: error: initialization of 'int (*)(const struct device *, struct net_if *, struct wifi_iface_status *)' from incompatible pointer type 'int (*)(const struct device *, struct wifi_iface_status *)' [-Wincompatible-pointer-types]
  651 |         .iface_status = esp_hosted_status,
```



### Regression

- [x] This is a regression.

### Steps to reproduce

_No response_

### Relevant log output

```shell

```

### Impact

Major – Severely degrades functionality; workaround is difficult or unavailable.

### Environment

_No response_

### Additional Context

_No response_

--- a/drivers/wifi/esp_hosted/esp_hosted_wifi.c
+++ b/drivers/wifi/esp_hosted/esp_hosted_wifi.c
@@ -492,8 +492,11 @@ static int esp_hosted_stats(const struct device *dev,
 }
 #endif
 
-static int esp_hosted_status(const struct device *dev, struct wifi_iface_status *status)
+static int esp_hosted_status(const struct device *dev, struct net_if *iface,
+			     struct wifi_iface_status *status)
 {
+	ARG_UNUSED(iface);
+
 	esp_hosted_data_t *data = dev->data;
 	CtrlMsg ctrl_msg = CtrlMsg_init_zero;
 	size_t itf = esp_hosted_get_iface(dev);

